internal/validation/validation.go

```diff
@@ -330,29 +330,26 @@ func ValidateUser(typesys *typesystem.TypeSystem, user string) error {
 		return fmt.Errorf("the 'user' field is malformed")
 	}
 
+	validObject := tuple.IsValidObject(user)
+	validUserset := tuple.IsObjectRelation(user)
+	userObject, userRelation := tuple.SplitObjectRelation(user)
+	userObjectType := tuple.GetType(userObject)
 	schemaVersion := typesys.GetSchemaVersion()
 
-	// the 'user' field must be an object (e.g. 'type:id') or object#relation (e.g. 'type:id#relation')
 	if typesystem.IsSchemaVersionSupported(schemaVersion) {
-		if !tuple.IsValidObject(user) && !tuple.IsObjectRelation(user) {
+		if !validObject && !validUserset {
 			return fmt.Errorf("the 'user' field must be an object (e.g. document:1) or an 'object#relation' or a typed wildcard (e.g. group:*)")
 		}
 
-		if tuple.IsObjectRelation(user) {
-			userObj, _ := tuple.SplitObjectRelation(user)
-
-			if tuple.IsTypedWildcard(userObj) {
-				return fmt.Errorf("the 'user' field cannot reference a typed wildcard in a userset value")
-			}
```

Comment:
This was replaced by adding a `*` in the userset regex

Reply:
This case is handled much earlier with just regex, correct?

Reply:
It is handled now by the `isValidUser` on line 329